src/views/List.jsx

```diff
@@ -1,16 +1,47 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 /** List component that displays items in a user's shopping cart  */
 export function List({ data }) {
+	const [filteredList, setFilteredList] = useState([]);
+	const [filterInput, setFilterInput] = useState('');
+
+	/* Use handler to change the state of filterInput  and convert all items to lowercase to facilitate a more thorough search */
+	const handleInput = (event) => {
+		const value = event.target.value.toLowerCase();
+		setFilterInput(value);
+		setFilteredList(
+			data.filter((item) => {
+				return item.name && item.name.toLowerCase().includes(value);
```

Comment:
In addition to `toLowerCase` and `includes`, consider adding `trim` (just before the `includes` method). Trimming takes away the white space from both ends of the string. As of this moment, it works fine if I type an item like "milk" and the matching item is rendered -- but if I type " milk " (white space on both ends), suddenly there is no match.

Reply:
great catch @arcan9!

Reply:
Good edge case to consider @arcan9.

Reply:
Love this idea!!